File: Java/백준/Gold/11729. 하노이 탑 이동 순서/하노이 탑 이동 순서.java
import java.util.*;

public class Main {
    public static StringBuilder sb = new StringBuilder();
    public static int count = 0;

    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();

        hanoi(n, 1, 2, 3);

        System.out.print(count + "\n" + sb.toString());
    }

    public static void hanoi(int board, int s, int v, int e) {
        count++;
        if (board == 1) {
            sb.append(s + " " + e + "\n");
            return;
        }

        hanoi(board - 1, s, e, v);

        sb.append(s + " " + e + "\n");

        hanoi(board - 1, v, s, e);
    }
}

File: Swift/백준/Gold/11729. 하노이 탑 이동 순서/하노이 탑 이동 순서.swift
import Foundation

let a = Int(readLine()!)!
var totalString: String = ""
var count = 0

func hanoi(_ board: Int, _ start: Int, _ via: Int, _ end: Int) {
    count += 1
    if board == 1 {
        totalString.append("\(start) \(end)\n")
        return
    }
    
    
    hanoi(board - 1, start, end, via)
    
    totalString.append("\(start) \(end)\n")
    
    hanoi(board - 1, via, start, end)
    
}

hanoi(a, 1, 2, 3)
print("\(count)\n\(totalString)")
